freqtrade/strategy/interface.py

```diff
@@ -637,35 +638,46 @@ def stop_loss_reached(self, current_rate: float, trade: Trade,
         # Initiate stoploss with open_rate. Does nothing if stoploss is already set.
         trade.adjust_stop_loss(trade.open_rate, stop_loss_value, initial=True)
 
+        # Stoploss rate and profit are calculated using open price in backtesting.
+        stop_rate = open or current_rate
+        stop_profit = trade.calc_profit_ratio(stop_rate)
+
         if self.use_custom_stoploss:
             stop_loss_value = strategy_safe_wrapper(self.custom_stoploss, default_retval=None
                                                     )(pair=trade.pair, trade=trade,
                                                       current_time=current_time,
-                                                      current_rate=current_rate,
-                                                      current_profit=current_profit)
+                                                      current_rate=stop_rate,
+                                                      current_profit=stop_profit)
             # Sanity check - error cases will return None
             if stop_loss_value:
                 # logger.info(f"{trade.pair} {stop_loss_value=} {current_profit=}")
-                trade.adjust_stop_loss(current_rate, stop_loss_value)
+                trade.adjust_stop_loss(stop_rate, stop_loss_value)
             else:
                 logger.warning("CustomStoploss function did not return valid stoploss")
 
         if self.trailing_stop:
             # trailing stoploss handling
             sl_offset = self.trailing_stop_positive_offset
-
-            # Make sure current_profit is calculated using high for backtesting.
-            high_profit = current_profit if not high else trade.calc_profit_ratio(high)
+            high_profit = trade.calc_profit_ratio(high or current_rate)
```

Comment:
the "prior" implementation (for this line!) was better - as it didn't recalculate profit if we're not in backtesting mode (and this doesn't really change how it works, anyway).